multy_core/src/ethereum/ethereum_transaction.cpp

```diff
@@ -411,7 +410,7 @@ void EthereumTransaction::serialize_to_stream(EthereumDataStream& stream, Serial
     list << m_gas;
     list << m_destination->address;
     list << m_destination->amount;
-    if (m_payload.is_set())
+    if (m_payload)
```

Comment:
Please also check that payload is non-null, but empty (len == 0 or data == nullptr).

Reply:
Please also add a run-time check in `operator<<(EthereumDataStream& stream, const BinaryData& binary_data)` that we are not trying to serialize BinaryData with zero `len` or nullptr `data`.